src/erl.rs

```diff
@@ -1380,3 +1486,163 @@ pub fn is_erlang_standard_library_module(name: &str) -> bool {
             | "zip"
     )
 }
+
+impl TypeAst {
+    pub fn to_erlang_type_def(&self) -> Document {
+        // reference : https://erlang.org/doc/reference_manual/typespec.html
+        match self {
+            TypeAst::Var { name, .. } | TypeAst::Hole { name, .. } => variable_name(name).to_doc(),
+            TypeAst::Tuple { elems, .. } => "{"
+                .to_doc()
+                .append(concat(
+                    elems
+                        .iter()
+                        .map(|e| e.to_erlang_type_def())
+                        .intersperse(", ".to_doc()),
+                ))
+                .append("}"),
+
+            TypeAst::Constructor {
+                module, name, args, ..
+            } => {
+                let args = if args.is_empty() {
+                    nil()
+                } else {
+                    concat(
+                        args.iter()
+                            .map(|a| a.to_erlang_type_def())
+                            .intersperse(", ".to_doc()),
+                    )
+                };
+                if let Some(mod_name) = &module {
+                    let t = mod_name
+                        .clone()
+                        .to_doc()
+                        .append(".")
+                        .append(atom(name.clone()));
+                    if args.is_nil() {
+                        t.append("()")
+                    } else {
+                        t.append("(").append(args).append(")")
+                    }
+                } else {
+                    match name.as_str() {
+                        // TODO: more builtins ? Result?
+                        "Int" => "integer()".to_doc(),
+                        "String" => "string()".to_doc(),
```

Comment:
This would be `unicode:unicode_binary()` as we use utf8 encoded binary strings, while Erlang uses character lists.

https://erlang.org/doc/man/unicode.html#type-unicode_binary